API routes for plants

AliNematollahi-HackYourFuture opened the pull request:
This PR introduces two new GET endpoints for the Plants API and adds documentation to help teammates understand and test the API.

Changes

Implemented GET /plants route → returns all plants

Implemented GET /plants/:id route → returns a plant by its ID

Updated README.md file documenting the API endpoints (Development Notes)

pankaj28843 commented on api/src/routers/plants.js:
```diff
@@ -0,0 +1,27 @@
+router.get("/", async (req, res) => {
+  try {
+    const plants = await db("plants").select("*");
+    res.json(plants);
```
Great work setting up your API routes! One thing I’d like you to consider is introducing a **serialization layer** between your database models and the data you expose to your API or UI.

For example, instead of directly returning objects from the database, you can write a function that maps those objects into **plain, application-friendly JavaScript objects**:

```js
const serializePlant = (dbPlant) => ({
  id: dbPlant.id,
  name: dbPlant.name,
  species: dbPlant.species,
  // ...map only the properties you actually want to expose
});
```

This practice gives you a clean separation between your **database schema** and your **application/domain model**. Without this, your UI or API clients become tightly coupled to your database structure. Over time, that makes refactoring your database much harder, because even small schema changes ripple through your entire codebase.

By introducing this mapping layer:

* ✅ You gain **flexibility**: changes in the DB won’t immediately break your UI.
* ✅ You improve **security & control**: you only expose safe, intentional fields.
* ✅ You support **maintainability**: future developers can understand the application shape of data without knowing the DB internals.

This is especially important as your project grows beyond simple prototypes into production-level applications.

---

### 📚 Further Reading & References

Here are an excellent blog post that dive deeper into this idea:

* [Martin Fowler on the Data Mapper pattern](https://martinfowler.com/eaaCatalog/dataMapper.html)— the classic explanation of why mapping between domain objects and persistence is important.
---

👉 In short: **think of serialization as giving your application its own language for data**, instead of directly speaking in “database.” It’s a habit that pays dividends as you work on larger, real-world systems.

LCurcuma commented on api/src/index.js:
```diff
@@ -9,6 +9,8 @@ const app = express();
 app.use(cors());
 app.use(express.json());
 
+app.use('/plants',plantsRouter)
```
the router doesn't work, because the router not imported. I recommend add "import plantsRouter from..."

LCurcuma commented on api/src/routers/plants.js:
```diff
@@ -0,0 +1,27 @@
+const plantsRouter = express.Router();
+
+// GET ALL plants
+router.get("/", async (req, res) => {
```
the "router" should be replaced with "plantsRouter". We have "const plantsRouter = express.Router();" not "const router = express.Router();"

LCurcuma commented on api/src/routers/plants.js:
```diff
@@ -0,0 +1,27 @@
+});
+
+// GET plant by id
+router.get("/:id", async (req, res) => {
```
also here. After the changes, the API works smoothly - I tested that :)

AliNematollahi-HackYourFuture commented:
* Implemented GET /plants/all route → returns all plants
* Implemented GET /plants/:id route → returns a plant by its ID
* Implemented GET /plants/limited-top-rating/:limit → returns limited top rating plants
* Implemented GET /plants → returns a paginated list of plants (Example: GET /plants?page=2&pageSize=10 → returns plants 11–20)
